yarn/src/main/scala/org/apache/spark/deploy/yarn/security/AMDelegationTokenRenewer.scala

```diff
@@ -193,8 +200,14 @@ private[yarn] class AMDelegationTokenRenewer(
       }
     }
     val nextSuffix = lastCredentialsFileSuffix + 1
+    // Time of next update should be late than renewal,
+    // which is around new token issue date + 0.8 * renew interval
+    val timeOfNextUpdate =
```

Comment:
This is also part of the logic that could be consolidated in `credentialManager.obtainCredentials`. Also, comments above don't really match what the code is doing.